src/test/java/org/opensearch/neuralsearch/processor/NeuralQueryProcessorTests.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.processor;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class NeuralQueryProcessorTests extends OpenSearchTestCase {
+
+    public void testFactory() throws Exception {
```

Comment:
please use naming convention from other tests - whatYouTesting_conditions_expectation, I think <conditions> may be optional sometimes, when it's obvious, but other two are must